src/data.rs

```diff
@@ -0,0 +1,119 @@
+use std::fmt;
+
+/// A byte buffer used for serialization to and from the plist data type.
+///
+/// You use it in types with derived `Serialize`/`Deserialize` traits.
+/// ## Examples
+///
+/// ```rust
+/// extern crate plist;
+/// # #[cfg(feature = "serde")]
+/// #[macro_use]
+/// extern crate serde_derive;
+///
+/// # #[cfg(feature = "serde")]
+/// # fn main() {
+/// #[derive(Deserialize, Serialize)]
+/// struct Info {
+///     blob: plist::Data,
+/// }
+///
+/// let actual = Info { blob: plist::Data::new(vec![1, 2, 3, 4]) };
+///
+/// let mut xml_byte_buffer: Vec<u8> = vec![];
+/// plist::to_writer_xml(&mut xml_byte_buffer, &actual)
+///     .expect("serialize into xml");
+///
+/// let expected: Info = plist::from_reader_xml(xml_byte_buffer.as_slice())
+///     .expect("deserialize from xml");
+///
+/// assert_eq!(actual.blob, expected.blob);
+/// # }
+/// #
+/// # #[cfg(not(feature = "serde"))]
+/// # fn main() {}
+/// ```
+#[derive(Clone, PartialEq)]
+pub struct Data {
+    inner: Vec<u8>,
+}
+
+impl Data {
+    /// Creates a new `Data` from vec of bytes.
+    pub fn new(inner: Vec<u8>) -> Self {
+        Data { inner }
+    }
+
+    /// Consumes itself and returns inner vec of bytes.
+    pub fn into_inner(self) -> Vec<u8> {
+        self.inner
+    }
+}
+
+impl From<Vec<u8>> for Data {
```

Comment:
Might be worth having `From<Data> for Vec<u8>` as well.

Reply:
I don't know. What is the best practice in rust? If we implement `From<Data> for Vec<u8>` do we still need `into_inner` method?